### DISCUSSION TITLE
How can I  get and update Scale subresource belog to k8s internal workload and CRD

### DISCUSSION THREAD

**Discussion by hzyfox** [kubernetes-sigs/kubebuilder#3486] [Support]
Is There a  client like  status clinet to get and update  Scale subresource  which belong to k8s internal workload (eg. Deployment, Statefulset) or User Defined Workalod  （eg. Open Krusie Statefulset）？

And how to build a Scale Client from scratch with golang ?

**Comment by camilamacedo86** [comment]
Hi @hzyfox,

The following example might help you out

```go
```go
package controllers

import (
	"context"
	"k8s.io/apimachinery/pkg/runtime"
	ctrl "sigs.k8s.io/controller-runtime"
	"sigs.k8s.io/controller-runtime/pkg/client"
	"k8s.io/api/autoscaling/v1"
)

func (r *YourReconciler) GetAndUpdateScale(ctx context.Context, namespace string, name string, desiredReplicas int32) error {
	scale := &v1.Scale{}
	scaleKey := client.ObjectKey{
		Namespace: namespace,
		Name:      name,
	}

	// Fetch current scale of the Deployment
	if err := r.Get(ctx, scaleKey, scale); err != nil {
		return err
	}

	// Update the replica count in the retrieved scale object
	scale.Spec.Replicas = desiredReplicas

	// Update the scale in the cluster
	if err := r.Update(ctx, scale); err != nil {
		return err
	}

	return nil
}
```

The same approach it seems to applies to `StatefulSet`, `DaemonSet`, and other scalable workloads. You just need to fetch and update the `Scale` subresource for the respective type.

For further details on using subresources with controller-runtime, you might want to check out the [official controller-runtime documentation](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client) and the [Kubebuilder book](https://book.kubebuilder.io/).
 
Suggestion: It might be helpful to review the content of this PR: [kubebuilder PR #3596](https://github.com/kubernetes-sigs/kubebuilder/pull/3596/files). It offers a comprehensive guide that might clarify how things operate in Kubebuider/Controller runtime.

I hope that helps you out
However, if you are looking for something outside of Kubebuilder or controller runtime than it might not be the best place. 

So, I am closing this one. However, if you think that it should be re-opened please feel free to do so,
